### PR TITLE
routing + usersPage + context

### PR DESCRIPTION
**Pull request by BelohvostovAlex**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
-import { FormContainer } from './components/Form';
+import React, { useState } from 'react';
+import { AppRouter } from './components/AppRouter';
+import { AppContext } from './context';
+
+export const App: React.FC = () => {
+  const [isAuth, setIsAuth] = useState(false);
+  const changeSetIsAuth = (val: boolean) => {
+    setIsAuth(val);
+  };
```
Comment (igorTT) [inline src/App.tsx]: This functionality hasn't been described in the issue.
https://github.com/BelohvostovAlex/webpack/issues/3

File: src/components/AppRouter/AppRouter.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React, { useContext } from 'react';
+import { Routes, Route, Navigate } from 'react-router-dom';
+import { AppContext } from '../../context';
+import { privateRoutes, publicRoutes } from '../../router';
+import { pathesEnum } from '../../router/interfaces';
+
+export const AppRouter: React.FC = () => {
+  const { isAuth } = useContext(AppContext);
+
+  return isAuth ? (
```
Comment (igorTT) [inline src/components/AppRouter/AppRouter.tsx]: This functionality hasn't been described in the issue.
https://github.com/BelohvostovAlex/webpack/issues/3

File: src/components/Form/FormContainer.tsx [hunk 3]
```diff
@@ -1,23 +1,29 @@
...
   const onSubmit = async () => {
     try {
       const response = await axios.post<IUser>('http://localhost:4000/api', {
         username: username,
         password: password,
       });
       setOutput(response.data);
+      setTimeout(() => {
+        changeSetIsAuth(true);
+      }, 500);
```
Comment (igorTT) [inline src/components/Form/FormContainer.tsx]: What is this?

File: src/pages/UsersPage/UsersPage.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  changeSetIsAuth,
+}) => {
+  return (
+    <div className="user-page">
+      <Button
+        btnClass="send sign-out__btn"
+        btnText="Sign out"
+        btnType={ButtonTypes.button}
+        onClick={() => changeSetIsAuth(false)}
+      />
```
Comment (igorTT) [inline src/pages/UsersPage/UsersPage.tsx]: I don't see this button in the requirements

File: src/pages/UsersPage/UsersPage.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+              />
+              <div>
+                {user.id}: {user.name}, user's email: {user.email}
+              </div>
+              <Button
+                btnClass="send delete__btn"
+                btnText="Delete"
+                btnType={ButtonTypes.button}
+                onClick={() => handleDelete(user.checked, user.id)}
+              />
```
Comment (igorTT) [inline src/pages/UsersPage/UsersPage.tsx]: don't pass 'delete button' in every row

File: src/pages/UsersPage/UsersPageContainer.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    try {
+      const { data } = await axios.get<IUser[]>(
+        'https://jsonplaceholder.typicode.com/users?_limit=10'
+      );
+      setUsers(
+        data.map((item) => {
+          return {
+            ...item,
+            checked: false,
+          };
```
Comment (igorTT) [inline src/pages/UsersPage/UsersPageContainer.tsx]: don't mutate the initial data.

File: src/pages/UsersPage/UsersPageContainer.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    }
+  };
+  const toggleChecked = (id: number) => {
+    setUsers([
+      ...users.map((user) => {
+        return user.id === id
+          ? { ...user, checked: !user.checked }
+          : { ...user };
+      }),
+    ]);
```
Comment (igorTT) [inline src/pages/UsersPage/UsersPageContainer.tsx]: Use a separate state to store checked information

**Review by igorTT — CHANGES_REQUESTED** [review]
Table should have only one 'Delete' button. After clicking this button all the users, those were checked should be removed from the table

File: src/App.tsx (hunk 1 above)
Comment (BelohvostovAlex) [reply src/App.tsx]: deleted

File: src/components/AppRouter/AppRouter.tsx (hunk 2 above)
Comment (BelohvostovAlex) [reply src/components/AppRouter/AppRouter.tsx]: deleted

File: src/components/Form/FormContainer.tsx (hunk 3 above)
Comment (BelohvostovAlex) [reply src/components/Form/FormContainer.tsx]: deleted

File: src/pages/UsersPage/UsersPage.tsx (hunk 4 above)
Comment (BelohvostovAlex) [reply src/pages/UsersPage/UsersPage.tsx]: deleted

File: src/pages/UsersPage/UsersPageContainer.tsx (hunk 7 above)
Comment (BelohvostovAlex) [reply src/pages/UsersPage/UsersPageContainer.tsx]: Will it be ok, if i just create a new state, add the key(checked) for each user and pass the new array as a props to the view component?

File: src/pages/UsersPage/UsersPage.tsx (hunk 5 above)
Comment (BelohvostovAlex) [reply src/pages/UsersPage/UsersPage.tsx]: done

File: src/pages/UsersPage/UsersPage.tsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+              </tr>
+            );
+          })}
+        </tbody>
+      </table>
+      <Button
+        btnClass="send users-page__btn"
+        btnText="delete"
+        btnType={ButtonTypes.button}
+        onClick={() => handleDelete()}
```
Comment (igorTT) [inline src/pages/UsersPage/UsersPage.tsx]: You can pass a plain function
```javascript
onClick={handleDelete}
```

File: src/pages/UsersPage/UsersPage.tsx [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+export const UsersPage: React.FC<UserPageProps> = ({
+  users,
+  handleDelete,
+  toggleChecked,
+}) => {
+  return (
+    <div className="users-page">
+      <table className="users-table">
+        <tbody>
+          {users.map((user) => {
```
Comment (igorTT) [inline src/pages/UsersPage/UsersPage.tsx]: Use destructuring

**Review by igorTT — CHANGES_REQUESTED** [review]

File: src/pages/UsersPage/UsersPage.tsx (hunk 9 above)
Comment (BelohvostovAlex) [reply src/pages/UsersPage/UsersPage.tsx]: done

File: src/pages/UsersPage/UsersPage.tsx (hunk 8 above)
Comment (BelohvostovAlex) [reply src/pages/UsersPage/UsersPage.tsx]: done

File: src/pages/UsersPage/UsersPageContainer.tsx [hunk 10]
```diff
@@ -17,38 +17,34 @@ export const UsersPageContainer: React.FC = () => {
...
-      };
-    });
-    setUsersWithCheck(newUsersWithCheck);
+  const getIsChecked = (id: number) => {
+    if (usersSet.has(id)) {
+      return true;
+    } else {
+      return false;
+    }
   };
```
Comment (igorTT) [inline src/pages/UsersPage/UsersPageContainer.tsx]: Replace with
```javascript
const getIsChecked = (id) => usersSet.has(id)
```

File: src/pages/UsersPage/UsersPageContainer.tsx [hunk 11]
```diff
@@ -1,11 +1,11 @@
 import axios from 'axios';
 import React, { useState, useEffect } from 'react';
-import { IUser, IUserWithCheck } from './interfaces';
+import { IUser } from './interfaces';
 import { UsersPage } from './UsersPage';
 
 export const UsersPageContainer: React.FC = () => {
   const [users, setUsers] = useState([] as IUser[]);
-  const [usersWithCheck, setUsersWithCheck] = useState([] as IUserWithCheck[]);
+  const [usersSet, setUsersSet] = useState(new Set());
```
Comment (igorTT) [inline src/pages/UsersPage/UsersPageContainer.tsx]: Don't use data type in the name. Rename with checkedUsersIDs, checkedIDs, checkedUsers or somethig else

File: src/pages/UsersPage/UsersPageContainer.tsx [hunk 12]
```diff
@@ -17,38 +17,34 @@ export const UsersPageContainer: React.FC = () => {
...
+    if (usersSet.has(id)) {
+      return true;
+    } else {
+      return false;
+    }
   };
 
   const deleteUser = () => {
-    setUsersWithCheck([...usersWithCheck.filter((user) => !user.checked)]);
+    setUsers([...users.filter((user) => !getIsChecked(user.id))]);
```
Comment (igorTT) [inline src/pages/UsersPage/UsersPageContainer.tsx]: Don't create a new array. Filter returns a new array by default

File: src/pages/UsersPage/UsersPageContainer.tsx [hunk 13]
```diff
@@ -17,38 +17,34 @@ export const UsersPageContainer: React.FC = () => {
...
-          ? { ...user, checked: !user.checked }
-          : { ...user };
-      }),
-    ]);
+    if (usersSet.has(id)) {
+      const arr = Array.from(usersSet).filter((item) => item !== id);
+      setUsersSet(new Set(arr));
+    } else {
+      setUsersSet((prevSet) => prevSet.add(id));
+    }
```
Comment (igorTT) [inline src/pages/UsersPage/UsersPageContainer.tsx]: Replace with
```javascript
setUsersSet((prev) => {
  const next = new Set(prev);

  if (next.has(id)) {
    next.delete(id);
  } else {
    next.add(id);
  }

  return next;
})
```

**Review by igorTT — CHANGES_REQUESTED** [review]

File: src/pages/UsersPage/UsersPageContainer.tsx (hunk 10 above)
Comment (BelohvostovAlex) [reply src/pages/UsersPage/UsersPageContainer.tsx]: changed, next time will use one-line

File: src/pages/UsersPage/UsersPageContainer.tsx (hunk 11 above)
Comment (BelohvostovAlex) [reply src/pages/UsersPage/UsersPageContainer.tsx]: noted, done

File: src/pages/UsersPage/UsersPageContainer.tsx (hunk 12 above)
Comment (BelohvostovAlex) [reply src/pages/UsersPage/UsersPageContainer.tsx]: done

File: src/pages/UsersPage/UsersPageContainer.tsx (hunk 13 above)
Comment (BelohvostovAlex) [reply src/pages/UsersPage/UsersPageContainer.tsx]: changed, noted

File: src/components/Form/FormContainer.tsx [hunk 14]
```diff
@@ -1,4 +1,4 @@
-import React, { useState } from 'react';
+import React, { useState, useContext } from 'react';
```
Comment (igorTT) [inline src/components/Form/FormContainer.tsx]: Delete unused import

**Review by igorTT — CHANGES_REQUESTED** [review]

File: src/components/Form/FormContainer.tsx (hunk 14 above)
Comment (BelohvostovAlex) [reply src/components/Form/FormContainer.tsx]: done

**Review by igorTT — APPROVED** [review]

**Pull request merged**